### ISSUE TITLE
Rollout.io long compilation time 

### ISSUE THREAD

**Issue by ekeren** [rollout/rollout.io-ios#11]
For project with huge file base it takes up to 20 seconds to run "Rollout Code Analyzer" build phase as part of the compilation.

**Comment by ekeren** [comment]
We are working on resolving this issue and remove the compilation time to zero, until we do we provide this current workaround:
For people who installed Rollout.io via installer:
```bash
if [ ! -f "${SRCROOT}/Rollout-ios-SDK/Rollout/RolloutSwizzlerDynamic.include" -o "${CONFIGURATION}" != "Debug" ]; then 
"${SRCROOT}/Rollout-ios-SDK/lib/tweaker" -k  <your-app-key>
fi
```
For people who installed Rollout.io via cocapods:
```bash
if [ ! -f "${PODS_ROOT}/Rollout.io/Rollout/RolloutSwizzlerDynamic.include" -o "${CONFIGURATION}" != "Debug" ] ; then
"${SRCROOT}/Pods/Rollout.io/lib/tweaker" -k <your-app-key>
fi
```

**Comment by imoldman** [comment]
+1

**Comment by rlopezdiez** [comment]
+1

**Comment by QuynhNguyen** [comment]
+1

**Comment by jspavlick** [comment]
+1

**Comment by sbhhbs** [comment]
+1

**Comment by ekeren** [comment]
Version 0.8.6+ resolves the problem of slow compile time. It also works with project that has Swift code included (although no swif support yet, see update in #14). 

Version 0.8.6+ also requires you to commit Rollout's auto generated code, this committed code is important for compile time and also for you to be able to checkout the exact version you have in production, to allow more consistency for debugging production issues. See the relevant [FAQ question] (http://support.rollout.io/v1.0/docs/rolloutio-installation-guide#what-are-the-files-under-auto_generated_code) 

Hey version 0.8.7 and 0.8.6 resolves the problem of slow compile time. It also works with project that has Swift code included (although no swif support yet, see update in #14). 

Version 0.8.6+ also requires you to commit Rollout's auto generated code, this committed code is important for compile time and also for you to be able to checkout the exact version you have in production. see the relevant [FAQ question] (http://support.rollout.io/v1.0/docs/rolloutio-installation-guide#what-are-the-files-under-auto_generated_code)

**Comment by coverback** [comment]
@ekeren But if the the auto code generation function is deterministic, and code can be easily recreated, why the need to commit it to source control?

**Comment by ekeren** [comment]
@coverback - you are basically correct, it is deterministic but per architecture, it also work in an optimistic way and doesn't remove functions that will be required by other archs. (or other preprocessor MACRO definitions that affect functions signature)
 
Crash reporters wise, It makes some sense to have the exact version like the one you submitted to appstore source code. (one that matched your submitted dsym file) 

Regarding the compile time committing this code actually generates a cache mechanism.  (which is also required by swift to workaround xcode issue with SWIFT http://openradar.appspot.com/radar?id=5330879249383424) 

We are actually checking another direction to improve the compile time and one that will allow patching all function on demand without waiting for the build support, but it will take some time to get it out there. 

It will be great if you can get back to us and give feedback on this solution, my concern that these files will be hard to work with? 

@silyevsk - FYI

**Comment by coverback** [comment]
@ekeren sorry for a late response. we're finally re-integrating the lib :)
I also think these files seem very cumbersome to work with, but we'll try.

How often is the generated code changing? I tried changing some signatures in the code and it didn't change at once, but I had to run release build to have all the architecture code in (otherwise one by one building for different devices produced more changes).

Also, I still didn't get why it can't be autogenerated each time you make a build, simply checking some hash for new changes. Release configuration would build all archs and it should work, no?

**Comment by ekeren** [comment]
Glad to say that this issue is fixed in latest versions (0.9+) but the current solution is far from perfect.
Compile time is fine, but there are too many files that rollout.io generates. Working with these files is, as mentioned by @coverback, cumbersome to work with... 

The good news is that we are working on solving this issue of dynamic files, get updates on  https://github.com/rollout/rollout.io-ios/issues/53 I beleive that once issue #53 is resolved rolllout.io integration and work model will be truely seamless.  Hope to get updates soon
